File: usedesk_sdk/src/main/java/ru/usedesk/sdk/data/framework/entity/request/InitChatRequest.java
package ru.usedesk.sdk.data.framework.entity.request;

import static ru.usedesk.sdk.domain.entity.Constants.VALUE_CURRENT_SDK;

public class InitChatRequest extends BaseRequest {

    public static final String TYPE = "@@server/chat/INIT";

    private String companyId;

    private String url;
    private Payload payload;

    public InitChatRequest() {
        super(TYPE);
        payload = new Payload();
    }

    public String getCompanyId() {
        return companyId;
    }

    public void setCompanyId(String companyId) {
        this.companyId = companyId;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public Payload getPayload() {
        return payload;
    }

    private class Payload {

        private String sdk;

        private Payload() {
            this.sdk = VALUE_CURRENT_SDK;
        }

        public String getSdk() {
            return sdk;
        }
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/data/repository/api/loader/socket/_entity/initchat/InitChatRequest.kt
package ru.usedesk.chat_sdk.data.repository.api.loader.socket._entity.initchat

import ru.usedesk.chat_sdk.data.repository.api.loader.socket._entity._extra.BaseRequest

internal class InitChatRequest(
    val token: String?,
    val companyId: String,
    val url: String,
    messageLimit: Int
) : BaseRequest(TYPE) {

    private val payload = Payload(messageLimit)

    class Payload(
        val messageLimit: Int
    ) {
        val sdk = VALUE_CURRENT_SDK
        val type = VALUE_TYPE_SDK
    }

    companion object {
        private const val TYPE = "@@server/chat/INIT"
        private const val VALUE_CURRENT_SDK = "android"
        private const val VALUE_TYPE_SDK = "sdk"
    }
}